src/utils/runtime/src/lib.rs

```diff
@@ -281,9 +292,67 @@ pub fn init_risingwave_logger(settings: LoggerSettings, registry: prometheus::Re
         });
     };
 
-    let filter = filter::Targets::new().with_target("aws_smithy_client::retry", Level::DEBUG);
+    // Tracing layer
+    if let Ok(endpoint) = std::env::var("RW_TRACING_ENDPOINT") {
+        use opentelemetry::{sdk, KeyValue};
+        use opentelemetry_semantic_conventions::resource;
+
+        let id = format!(
+            "{}-{}",
+            hostname::get()
+                .ok()
+                .and_then(|o| o.into_string().ok())
+                .unwrap_or_default(),
+            std::process::id()
+        );
+
+        let otel_tracer = {
+            let runtime = tokio::runtime::Builder::new_multi_thread()
+                .enable_all()
+                .thread_name("risingwave-otel")
+                .worker_threads(2)
+                .build()
+                .unwrap();
+            let runtime = Box::leak(Box::new(runtime));
+
+            let _entered = runtime.enter();
+
+            opentelemetry_otlp::new_pipeline()
+                .tracing()
+                .with_exporter(
+                    opentelemetry_otlp::new_exporter()
+                        .tonic()
+                        .with_endpoint(endpoint),
+                )
+                .with_trace_config(sdk::trace::config().with_resource(sdk::Resource::new([
+                    KeyValue::new(
+                        resource::SERVICE_NAME,
+                        // TODO(bugen): better service name
+                        // https://github.com/jaegertracing/jaeger-ui/issues/336
+                        format!("{}-{}", settings.name, id),
```

Comment:
Typically for web services, the nodes with the same service type (for HA purposes) should have the same name, so they'll not be distinguishable in Jaeger UI. However I find it not that fit our usages so I add the unique ID here.